cylc/flow/platforms.py

```diff
@@ -548,5 +564,5 @@ def get_random_platform_for_install_target(
 
 def get_localhost_install_target() -> str:
     """Returns the install target of localhost platform"""
-    localhost = platform_from_name()
+    localhost = get_platform()
```

Comment:
This is just changing it back to how it was before; both identical as can be seen from `def get_platform(task_conf = None, task_id = UNKNOWN_TASK):`
https://github.com/cylc/cylc-flow/blob/bc72520ce03b09d492d3870fa8e2bfc76aa79da7/cylc/flow/platforms.py#L85-L87

The reason I had previously changed it to `platform_from_name()` was because mypy was complaining that `localhost` might be `None` due to the `Optional` return of `get_platform()`. However this now is fixed by the use of `@overload` to confirm that if `task_conf` is `None`, the return value will definitely be a dictionary
https://github.com/cylc/cylc-flow/blob/bc72520ce03b09d492d3870fa8e2bfc76aa79da7/cylc/flow/platforms.py#L43-L47

Reply:
I've just found out about a nasty omission in mypy's checking of `@overload`ed function implementations: https://github.com/python/mypy/issues/9503